pandas/core/reshape/concat.py

```diff
@@ -477,6 +477,7 @@ def _sanitize_mixed_ndim(
 
         else:
             name = getattr(obj, "name", None)
+            name_none_flag = False
```

Comment:
This name isn't accurate - will be False even in cases where `name` is `None`. Can you rename, I'd suggest `rename_columns`.